pkg/devcontainer/setup.go

```diff
@@ -95,28 +104,90 @@ func (r *runner) setupContainer(
 	cancelCtx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	errChan := make(chan error, 1)
+	errChan := make(chan error, 2)
 	go func() {
-		defer r.Log.Debugf("Done executing up command")
+		defer r.Log.Debugf("Done executing ssh server helper command")
 		defer cancel()
 
 		writer := r.Log.Writer(logrus.InfoLevel, false)
 		defer writer.Close()
 
-		r.Log.Debugf("Run command in container: %s", command)
-		err = r.Driver.CommandDevContainer(cancelCtx, r.ID, "root", command, stdinReader, stdoutWriter, writer)
-		if err != nil {
+		r.Log.Debugf("Run command in container: %s", sshCmd)
+		err = r.Driver.CommandDevContainer(cancelCtx, r.ID, "root", sshCmd, stdinReader, stdoutWriter, writer)
+		if err != nil && !errors.Is(err, context.Canceled) && !strings.Contains(err.Error(), "signal: ") {
 			errChan <- fmt.Errorf("executing container command: %w", err)
 		} else {
 			errChan <- nil
 		}
 	}()
 
+	// create pipes
+	stdoutReader2, stdoutWriter2, err := os.Pipe()
+	if err != nil {
+		return nil, err
+	}
+	stdinReader2, stdinWriter2, err := os.Pipe()
+	if err != nil {
+		return nil, err
+	}
+	defer stdoutWriter2.Close()
+	defer stdinWriter2.Close()
+
+	go func() {
```

Comment:
we're doing this whole setup once on the machine and once in the container. Do you think it's worth refactoring into a separate pkg or function?